content/docs/guides/promo-expressions.mdx

```diff
@@ -0,0 +1,84 @@
+---
+title: 'Rules-Based Promotion Expressions'
+section: 'Guides'
+priority: 4
+hidden: false
+---
+
+Writing custom rule-base expressions for your order promotions may seem intimidating at first, but once you get the hang of it you'll realize there are virtually limitless possibilities which provide the flexibility to meet almost any business need. In fact, custom order promotions are a great example of how our core value *Flexibility over Features* informs everything we build (and don't build) into the platform. We've found that allowing developers to define their own custom if-then statements is incredibly robust and solves more problems than providing a limited set of non-customizable promotions out-of-the-box.
+
+## Where should I start?
+
+Every promotion requires two rule expressions, an `EligibleExpression`, which evaluates the current state of the order and returns true or false indicating whether the promotion can be applied to the order, and a `ValueExpression`, which evaluates the order and returns a monetary value, which is then subtracted from the order subtotal.
+
+**There are several properties that you can include in both expressions to build your custom promotion:**\
+`order` supports the same properties as the Order model returned from /orders API endpoints, including xp.\
+`items` supports a total of four functions:
+ - `items.any()` (true if any item on the order matches filter)
+ - `items.all()` (true if all item matches filter)
+ - `items.quantity()`(compare result to a number)
+ - `items.total()` (compare result to a dollar amount)
+
+**The following operators are all supported in rule engine expressions:**
+-  Comparison: `=` ,  `<` ,  `>`,  `<=`,  `>=`  
+-  Logical: `and`,  `or`  and  `not`
+-  Mathematical:`+`,  `-`,`*`, `/` and `%` 
+
+##  Can you give me some examples?
+
+**$10 off any order greater than $50:**\
+`EligibleExpression`: *"order.Subtotal > 50"*\
+`ValueExpression`: *"10"*
+
+**Free Shipping when you spend $60:**\
+`EligibleExpression`: *"order.Subtotal >= 60"*\
+`ValueExpression`: *"order.ShippingCost"*
+
+**BOGO (limited to 1 free item):**
+`EligibleExpression`: *"items.quantity(ProductID = 'ABC') > 1"*\
+`ValueExpression`: *"items.total(ProductID = 'ABC') / items.quantity(ProductID = 'ABC')"*\
+
+**$5 off the order total when any line item has a given product ID:**\
+`EligibleExpression`: *"items.any(ProductID = '123')"*\
+`ValueExpression`: *"5"*
+
+**15% off all line items with a product assigned to a given CategoryID:**\
+`EligibleExpression`: *"items.any(product.incategory('Bikes'))"*\
+`ValueExpression`: *"items.total(product.incategory('Bikes')) * .15"*
```

Comment:
I think it'd be good to mention the roadmap feature `product.inparentcategory` and emphasize that a product must be directly assigned to the category in order for this to evaluate correctly. Might prevent some headaches

Reply:
I'll mention it explicitly down in the roadmap section that the current functionality only supports direct assignments